// next@16.1.6/dist/server/route-modules/app-page/module.d_20260223224341.ts
import type { AppPageRouteDefinition } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/route-definitions/app-page-route-definition';
import type RenderResult from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/render-result';
import type { RenderOpts } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/app-render/types';
import type { NextParsedUrlQuery } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/request-meta';
import type { LoaderTree } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/app-dir-module';
import type { PrerenderManifest } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/build';
import { renderToHTMLOrFlight, type AppSharedContext } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/app-render/app-render';
import { RouteModule, type RouteModuleOptions, type RouteModuleHandleContext } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/route-modules/route-module';
import * as vendoredContexts from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/route-modules/app-page/vendored/contexts/entrypoints';
import type { BaseNextRequest, BaseNextResponse } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/base-http';
import type { ServerComponentsHmrCache } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/response-cache';
import type { OpaqueFallbackRouteParams } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/request/fallback-params';
import type { DeepReadonly } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/shared/lib/deep-readonly';
/**
 * The AppPageModule is the type of the module exported by the bundled app page
 * module.
 */
export type AppPageModule = typeof import('.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/build/templates/app-page');
type AppPageUserlandModule = {
    /**
     * The tree created in next-app-loader that holds component segments and modules
     */
    loaderTree: LoaderTree;
};
export interface AppPageRouteHandlerContext extends RouteModuleHandleContext {
    page: string;
    query: NextParsedUrlQuery;
    fallbackRouteParams: OpaqueFallbackRouteParams | null;
    renderOpts: RenderOpts;
    serverComponentsHmrCache?: ServerComponentsHmrCache;
    sharedContext: AppSharedContext;
}
export type AppPageRouteModuleOptions = RouteModuleOptions<AppPageRouteDefinition, AppPageUserlandModule>;
export declare class AppPageRouteModule extends RouteModule<AppPageRouteDefinition, AppPageUserlandModule> {
    private matchers;
    match(pathname: string, prerenderManifest: DeepReadonly<PrerenderManifest>): {
        readonly dataRoute: string | null;
        readonly dataRouteRegex: string | null;
        readonly experimentalBypassFor?: readonly ({
            readonly type: "header" | "cookie" | "query";
            readonly key: string;
            readonly value?: string | undefined;
        } | {
            readonly type: "host";
            readonly key?: undefined | undefined;
            readonly value: string;
        })[] | undefined;
        readonly fallback: string | boolean | null;
        readonly fallbackRevalidate: import(".pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/cache-control").Revalidate | undefined;
        readonly fallbackExpire: number | undefined;
        readonly fallbackHeaders?: {
            readonly [x: string]: string;
        } | undefined;
        readonly fallbackStatus?: number | undefined;
        readonly fallbackRootParams: readonly string[] | undefined;
        readonly fallbackRouteParams: readonly {
            readonly paramName: string;
            readonly paramType: import(".pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/shared/lib/app-router-types").DynamicParamTypes;
        }[] | undefined;
        readonly fallbackSourceRoute: string | undefined;
        readonly prefetchDataRoute: string | null | undefined;
        readonly prefetchDataRouteRegex: string | null | undefined;
        readonly routeRegex: string;
        readonly experimentalPPR: boolean | undefined;
        readonly renderingMode: import(".pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/build/rendering-mode").RenderingMode | undefined;
        readonly allowHeader: readonly string[];
    } | null;
    render(req: BaseNextRequest, res: BaseNextResponse, context: AppPageRouteHandlerContext): Promise<RenderResult>;
    private pathCouldBeIntercepted;
    getVaryHeader(resolvedPathname: string, interceptionRoutePatterns: RegExp[]): string;
}
declare const vendored: {
    'react-rsc': typeof import(".pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/route-modules/app-page/vendored/rsc/entrypoints") | undefined;
    'react-ssr': typeof import(".pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/route-modules/app-page/vendored/ssr/entrypoints") | undefined;
    contexts: typeof vendoredContexts;
};
export { renderToHTMLOrFlight, vendored };
export default AppPageRouteModule;
