// next@16.1.6/dist/lib/metadata/metadata.d_20260223224337.ts
import React from 'react';
import type { GetDynamicParamFromSegment } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/app-render/app-render';
import type { LoaderTree } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/app-dir-module';
import type { SearchParams } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/request/search-params';
import { type MetadataErrorType } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/lib/metadata/resolve-metadata';
import type { MetadataContext } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/lib/metadata/types/resolvers';
import type { WorkStore } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/app-render/work-async-storage.external';
export declare function createMetadataComponents({ tree, pathname, parsedQuery, metadataContext, getDynamicParamFromSegment, errorType, workStore, serveStreamingMetadata, }: {
    tree: LoaderTree;
    pathname: string;
    parsedQuery: SearchParams;
    metadataContext: MetadataContext;
    getDynamicParamFromSegment: GetDynamicParamFromSegment;
    errorType?: MetadataErrorType | 'redirect';
    workStore: WorkStore;
    serveStreamingMetadata: boolean;
}): {
    Viewport: React.ComponentType;
    Metadata: React.ComponentType;
    MetadataOutlet: React.ComponentType;
};
