// next@16.1.6/dist/server/dev/static-paths-worker.d_20260223224348.ts
import type { NextConfigComplete } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/config-shared';
import '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/require-hook';
import '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/node-environment';
import type { StaticPathsResult } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/build/static-paths/types';
import type { IncrementalCache } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/incremental-cache';
import { type ExperimentalPPRConfig } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/experimental/ppr';
type RuntimeConfig = {
    pprConfig: ExperimentalPPRConfig | undefined;
    configFileName: string;
    cacheComponents: boolean;
};
export declare function loadStaticPaths({ dir, distDir, pathname, config, httpAgentOptions, locales, defaultLocale, isAppPath, page, isrFlushToDisk, fetchCacheKeyPrefix, cacheMaxMemorySize, requestHeaders, cacheHandler, cacheHandlers, cacheLifeProfiles, nextConfigOutput, buildId, authInterrupts, sriEnabled, }: {
    dir: string;
    distDir: string;
    pathname: string;
    config: RuntimeConfig;
    httpAgentOptions: NextConfigComplete['httpAgentOptions'];
    locales?: readonly string[];
    defaultLocale?: string;
    isAppPath: boolean;
    page: string;
    isrFlushToDisk?: boolean;
    fetchCacheKeyPrefix?: string;
    cacheMaxMemorySize: number;
    requestHeaders: IncrementalCache['requestHeaders'];
    cacheHandler?: string;
    cacheHandlers?: NextConfigComplete['cacheHandlers'];
    cacheLifeProfiles?: {
        [profile: string]: import('.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/use-cache/cache-life').CacheLife;
    };
    nextConfigOutput: 'standalone' | 'export' | undefined;
    buildId: string;
    authInterrupts: boolean;
    sriEnabled: boolean;
}): Promise<StaticPathsResult>;
export {};
