// next@16.1.6/dist/server/config.d_20260223224330.ts
import { PHASE_DEVELOPMENT_SERVER, type PHASE_PRODUCTION_SERVER, type PHASE_TYPE } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/shared/lib/constants';
import type { ExperimentalConfig, NextConfigComplete, NextConfig, NextConfigRuntime } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/config-shared';
export { normalizeConfig } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/config-shared';
export type { DomainLocale, NextConfig } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/config-shared';
export declare function warnOptionHasBeenDeprecated(config: NextConfig, nestedPropertyKey: string, reason: string, silent: boolean): boolean;
export declare function warnOptionHasBeenMovedOutOfExperimental(config: NextConfig, oldExperimentalKey: string, newKey: string, configFileName: string, silent: boolean): NextConfig;
type LoadConfigOptions = {
    customConfig?: object | null;
    rawConfig?: boolean;
    silent?: boolean;
    reportExperimentalFeatures?: (configuredExperimentalFeatures: ConfiguredExperimentalFeature[]) => void;
    reactProductionProfiling?: boolean;
    debugPrerender?: boolean;
};
export default function loadConfig(phase: typeof PHASE_DEVELOPMENT_SERVER, dir: string, opts?: LoadConfigOptions): Promise<NextConfigComplete>;
export default function loadConfig(phase: typeof PHASE_PRODUCTION_SERVER | typeof PHASE_DEVELOPMENT_SERVER, dir: string, opts?: LoadConfigOptions): Promise<NextConfigRuntime | NextConfigComplete>;
export default function loadConfig(phase: PHASE_TYPE, dir: string, opts?: LoadConfigOptions): Promise<NextConfigComplete>;
export type ConfiguredExperimentalFeature = {
    key: keyof ExperimentalConfig;
    value: ExperimentalConfig[keyof ExperimentalConfig];
    reason?: string;
};
