// next@16.1.6/dist/shared/lib/hooks-client-context.shared-runtime.d_20260223224331.ts
import type { Params } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/request/params';
import { ReadonlyURLSearchParams } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/client/components/readonly-url-search-params';
export declare const SearchParamsContext: import("react").Context<URLSearchParams | null>;
export declare const PathnameContext: import("react").Context<string | null>;
export declare const PathParamsContext: import("react").Context<Params | null>;
export type InstrumentedPromise<T> = Promise<T> & {
    status: 'fulfilled';
    value: T;
    displayName: string;
};
export type NavigationPromises = {
    pathname: InstrumentedPromise<string>;
    searchParams: InstrumentedPromise<ReadonlyURLSearchParams>;
    params: InstrumentedPromise<Params>;
    selectedLayoutSegmentPromises?: Map<string, InstrumentedPromise<string | null>>;
    selectedLayoutSegmentsPromises?: Map<string, InstrumentedPromise<string[]>>;
};
export declare const NavigationPromisesContext: import("react").Context<NavigationPromises | null>;
export declare function createDevToolsInstrumentedPromise<T>(displayName: string, value: T): InstrumentedPromise<T>;
export { ReadonlyURLSearchParams };
