// next@16.1.6/dist/server/lib/dev-bundler-service.d_20260223224337.ts
import type { IncomingMessage } from 'http';
import type { DevBundler } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/router-utils/setup-dev-bundler';
import type { WorkerRequestHandler } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/types';
import { LRUCache } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/lib/lru-cache';
import { type HmrMessageSentToBrowser, type NextJsHotReloaderInterface } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/dev/hot-reloader-types';
/**
 * The DevBundlerService provides an interface to perform tasks with the
 * bundler while in development.
 */
export declare class DevBundlerService {
    private readonly bundler;
    private readonly handler;
    appIsrManifestInner: InstanceType<typeof LRUCache<boolean>>;
    close: NextJsHotReloaderInterface['close'];
    setCacheStatus: NextJsHotReloaderInterface['setCacheStatus'];
    setReactDebugChannel: NextJsHotReloaderInterface['setReactDebugChannel'];
    sendErrorsToBrowser: NextJsHotReloaderInterface['sendErrorsToBrowser'];
    constructor(bundler: DevBundler, handler: WorkerRequestHandler);
    ensurePage: typeof this.bundler.hotReloader.ensurePage;
    logErrorWithOriginalStack: (err: unknown, type?: "unhandledRejection" | "uncaughtException" | "warning" | "app-dir") => void;
    getFallbackErrorComponents(url?: string): Promise<void>;
    getCompilationError(page: string): Promise<any>;
    revalidate({ urlPath, revalidateHeaders, opts: revalidateOpts, }: {
        urlPath: string;
        revalidateHeaders: IncomingMessage['headers'];
        opts: any;
    }): Promise<{}>;
    get appIsrManifest(): Record<string, boolean>;
    setIsrStatus(key: string, value: boolean | undefined): void;
    sendHmrMessage(message: HmrMessageSentToBrowser): void;
}
