// next@16.1.6/dist/build/segment-config/app/app-segments.d_20260223224346.ts
import type { Params } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/request/params';
import type { AppPageRouteModule } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/route-modules/app-page/module.compiled';
import type { AppRouteRouteModule } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/route-modules/app-route/module.compiled';
import { type AppSegmentConfig } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/build/segment-config/app/app-segment-config';
import type { DynamicParamTypes } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/shared/lib/app-router-types';
type GenerateStaticParams = (options: {
    params?: Params;
}) => Promise<Params[]>;
export type AppSegment = {
    name: string;
    paramName: string | undefined;
    paramType: DynamicParamTypes | undefined;
    filePath: string | undefined;
    config: AppSegmentConfig | undefined;
    generateStaticParams: GenerateStaticParams | undefined;
};
/**
 * Collects the segments for a given route module.
 *
 * @param components the loaded components
 * @returns the segments for the route module
 */
export declare function collectSegments(routeModule: AppRouteRouteModule | AppPageRouteModule): Promise<AppSegment[]> | AppSegment[];
export {};
