// next@16.1.6/dist/client/components/client-page.d_20260223224344.ts
import type { ParsedUrlQuery } from 'querystring';
import type { Params } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/server/request/params';
/**
 * When the Page is a client component we send the params and searchParams to this client wrapper
 * where they are turned into dynamically tracked values before being passed to the actual Page component.
 *
 * additionally we may send promises representing the params and searchParams. We don't ever use these passed
 * values but it can be necessary for the sender to send a Promise that doesn't resolve in certain situations.
 * It is up to the caller to decide if the promises are needed.
 */
export declare function ClientPageRoot({ Component, serverProvidedParams, }: {
    Component: React.ComponentType<any>;
    serverProvidedParams: null | {
        searchParams: ParsedUrlQuery;
        params: Params;
        promises: Array<Promise<any>> | null;
    };
}): import("react/jsx-runtime").JSX.Element;
