// next@16.1.6/dist/client/components/layout-router.d_20260223224342.ts
import type { ErrorComponent } from '.pnpm/next@16.1.6_@babel+core@7.29.0_react-dom@19.2.3_react@19.2.3__react@19.2.3/node_modules/next/dist/client/components/error-boundary';
import React from 'react';
/**
 * OuterLayoutRouter handles the current segment as well as <Offscreen> rendering of other segments.
 * It can be rendered next to each other with a different `parallelRouterKey`, allowing for Parallel routes.
 */
export default function OuterLayoutRouter({ parallelRouterKey, error, errorStyles, errorScripts, templateStyles, templateScripts, template, notFound, forbidden, unauthorized, segmentViewBoundaries, }: {
    parallelRouterKey: string;
    error: ErrorComponent | undefined;
    errorStyles: React.ReactNode | undefined;
    errorScripts: React.ReactNode | undefined;
    templateStyles: React.ReactNode | undefined;
    templateScripts: React.ReactNode | undefined;
    template: React.ReactNode;
    notFound: React.ReactNode | undefined;
    forbidden: React.ReactNode | undefined;
    unauthorized: React.ReactNode | undefined;
    segmentViewBoundaries?: React.ReactNode;
}): React.ReactNode[];
